rest.c

```diff
@@ -1802,10 +1802,12 @@ rest_api (req_handle handle, int flags, const char *rpath, const char *path,
                                  remote_user, remote_addr);
         if ((flags & FLAGS_METHOD_HEAD) && resp)
         {
-            g_free (resp);
-            resp = g_strdup_printf ("Status: %d\r\n"
-                        "Content-Type: %s\r\n\r\n", HTTP_CODE_OK,
-                        flags & FLAGS_RESTCONF ? "application/yang-data+json" : "application/json");
+            char *ptr = strstr (resp, "\r\n\r\n");
```

Comment:
better to update rest_api_get to not insert the data at the end of processing if flags & HEAD

and can we have a unit test please